pkg/filesystem/subvolume.go

```diff
@@ -407,6 +518,33 @@ func getOmapKey(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNam
 	return omapkey
 }
 
+// getSnapOmapKey gets the omap key and value details for a given snapshot.
+// Deletion of omap object required the snapshotName which is of format
+// csi.snap.snapid.
+// similarly to delete of omap key requires csi.snap.ompakey, where
+// omapkey is the snapshotcontent name which is extracted the omap object.
+func getSnapOmapKey(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, cephClusterNamespace, snap, fs string) string {
+
+	poolName, err := getMetadataPoolName(ctx, clientsets, operatorNamespace, cephClusterNamespace, fs)
+	if err != nil || poolName == "" {
+		logging.Fatal(fmt.Errorf("pool name not found: %q", err))
+	}
+	snapomapval, _ := getSnapOmapVal(snap)
+
+	args := []string{"getomapval", snapomapval, "csi.snapname", "-p", poolName, "--namespace", "csi", "/dev/stdout"}
+	cmd := "rados"
+	snapshotcontentname, err := runCommand(ctx, clientsets, operatorNamespace, cephClusterNamespace, cmd, args)
+	if err != nil || snapshotcontentname == "" {
+		logging.Info("No snapshot content found for snapshot %s: %s", snap, err)
+		return ""
+	}
```

Comment:
am not sure this check is correct, we can consider the error !=nil as not found right?

Reply:
yes correct.

Reply:
This looks like a wrong assumption to consider `err!=nil` as not found and continue , can you please check this again

Reply:
Having this assumption will leave a stale entry in the omap, can you please check, if there is any error when running this command it should be logged as Fatal not just info and return empty struct which will skip omap deletion operation.

Reply:
okay checked that, added the required change

Reply:
@Madhu-1 please do review again.